src/lib/components/layout/QLayout.svelte

```diff
@@ -22,116 +20,131 @@
-        ? rightRailbarWidth
-        : `${rightRailbarWidth}px`,
-      "--left-drawer-width": isNaN(Number(leftDrawerWidth))
-        ? leftDrawerWidth
-        : `${leftDrawerWidth}px`,
-      "--right-drawer-width": isNaN(Number(rightDrawerWidth))
-        ? rightDrawerWidth
-        : `${rightDrawerWidth}px`,
-    },
-    userStyles
-  );
+  const isNumeric = (measure: string | number) =>
+    typeof measure === "number" || !isNaN(Number(measure));
 
-  $: classes = createClasses(["q-layout", userClasses]);
+  const getStyleValue = (measure: string | number) =>
+    isNumeric(measure) ? `${measure}px` : measure;
+
+  Q.classes("q-layout", {
+    classes: [props.class],
+  });
 
   function prepareCtx(viewProp: typeof view) {
     const [top, middle, bottom] = viewProp!.split(" ");
-    const header: AppbarContext | undefined = $$slots.header
+    const headerCtx: AppbarContext | undefined = header
       ? {
           display: true,
           fixed: top.includes("H"),
         }
       : undefined;
-    const footer: AppbarContext | undefined = $$slots.footer
+    const footerCtx: AppbarContext | undefined = footer
       ? {
           display: true,
           fixed: bottom.includes("F"),
         }
       : undefined;
-    const drawerLeft: DrawerContext = {
+    const drawerLeftCtx: DrawerContext = {
       offset: {
-        top: $$slots.header && top[0].toLowerCase() === "h",
-        bottom: $$slots.footer && bottom[0].toLowerCase() === "f",
+        top: !!header && top[0].toLowerCase() === "h",
+        bottom: !!footer && bottom[0].toLowerCase() === "f",
       },
       fixed: [top[0], middle[0], bottom[0]].includes("L"),
-      railbar: $$slots.railbarLeft === true,
-      drawer: $$slots.drawerLeft === true,
+      railbar: !!railbarLeft,
+      drawer: !!drawerLeft,
       overlay: false,
     };
-    const drawerRight: DrawerContext = {
+    const drawerRightCtx: DrawerContext = {
       offset: {
-        top: $$slots.header && top[2].toLowerCase() === "h",
-        bottom: $$slots.footer && bottom[2].toLowerCase() === "f",
+        top: !!header && top[2].toLowerCase() === "h",
+        bottom: !!footer && bottom[2].toLowerCase() === "f",
       },
       fixed: [top[2], middle[2], bottom[2]].includes("R"),
-      railbar: $$slots.railbarRight === true,
-      drawer: $$slots.drawerRight === true,
+      railbar: !!railbarRight,
+      drawer: !!drawerRight,
       overlay: false,
     };
 
     return {
-      header,
-      footer,
-      drawerLeft,
-      drawerRight,
+      header: headerCtx,
+      footer: footerCtx,
+      drawerLeft: drawerLeftCtx,
+      drawerRight: drawerRightCtx,
     };
   }
 
-  let ctx: LayoutContext = writable(prepareCtx(view));
-  $: $ctx = prepareCtx(view);
-  setContext("layout", ctx);
+  const ctx = new QContext<LayoutContext>("layout", prepareCtx(view));
+
+  $effect(() => {
+    untrack(() => ctx);
+    ctx.update(prepareCtx(view));
```

Comment:
Thise two lines should become `untrack(() => ctx).update(prepareCtx(view));`.
Could you try it?